backends/database.js

```diff
@@ -132,7 +163,10 @@ module.exports = class Database extends Observable {
     }
 
     // required
-    if (!!field.required && !(field.required instanceof Function)) {
+    if (
+      (!!field.required && !(field.required instanceof Function)) ||
+      field.fieldtype === 'Currency'
```

Comment:
question: is it necessary to force a currency field to be non-nullable at the frappejs level rather than being handled at the application end?

Reply:
Yup, primarily cause they trip up calculations in unexpected ways. `null` is also an invalid input to `pesa` by design.

The problems arising cause of this has been taken care of using [`getPreDefaultValues`](https://github.com/frappe/frappejs/pull/153/files#diff-3068bf2cf3bbed9d15ad34016207cc7399c97015b1734d99575317e7edf203e0R713).